### PR TITLE
Tests that check that activities started from an external intent create ODK dirs were added.

### PR DESCRIPTION
**Pull request by sanovikov71**
Tests that check that activities started from an external intent create ODK dirs were added.

Change-Id: If8a24bbcace662f437ce0442a651af92a37b51fa

### CODE REVIEW

**Comment by sanovikov71** [comment]
Some tests that can be useful to solve #545.

File: collect_app/src/main/java/org/odk/collect/android/activities/FormEntryActivity.java [hunk 1]
```diff
@@ -355,7 +355,7 @@ public void onClick(View v) {
             if (intent != null) {
                 Uri uri = intent.getData();
 
-                if (getContentResolver().getType(uri).equals(InstanceColumns.CONTENT_ITEM_TYPE)) {
+                if (null != uri && getContentResolver().getType(uri).equals(InstanceColumns.CONTENT_ITEM_TYPE)) {
```
Comment (lognaturel) [inline collect_app/src/main/java/org/odk/collect/android/activities/FormEntryActivity.java]: You'll see that we typically put the thing that changes (`uri` in this case) before the thing that doesn't (`null` in this case) when doing comparisons. It's a convention I didn't realize we had but you'll see it's very consistent across the code base!

File: collect_app/src/main/java/org/odk/collect/android/activities/InstanceChooserList.java [hunk 2]
```diff
@@ -68,7 +68,7 @@ public void onCreate(Bundle savedInstanceState) {
         setContentView(R.layout.chooser_list_layout);
 
         String order;
-        if (getIntent().getStringExtra(ApplicationConstants.BundleKeys.FORM_MODE).equalsIgnoreCase(ApplicationConstants.FormModes.EDIT_SAVED)) {
+        if ((ApplicationConstants.FormModes.EDIT_SAVED).equalsIgnoreCase(getIntent().getStringExtra(ApplicationConstants.BundleKeys.FORM_MODE))) {
```
Comment (lognaturel) [inline collect_app/src/main/java/org/odk/collect/android/activities/InstanceChooserList.java]: This is an unrelated change so please remove it from this pull request! 

Interestingly, I think it's the same kind of thing as the null check above -- we tend to put the unchanging thing after the changing one in tests but it looks like you have a different style. There's nothing wrong with the way you do it but for consistency it would be good to stick to one convention. 😊

**Review by lognaturel — CHANGES_REQUESTED** [review]
Great initiative! These will be really good tests to have in place.

I made a couple of quick stylistic comments.

File: collect_app/src/main/java/org/odk/collect/android/activities/FormEntryActivity.java (hunk 1 above)
Comment (sanovikov71) [reply collect_app/src/main/java/org/odk/collect/android/activities/FormEntryActivity.java]: done

File: collect_app/src/main/java/org/odk/collect/android/activities/InstanceChooserList.java (hunk 2 above)
Comment (sanovikov71) [reply collect_app/src/main/java/org/odk/collect/android/activities/InstanceChooserList.java]: I made these changes because otherwise there was a NullPointerException in InstanceChooserListTest.

The main difference between CONSTANT.equals(object) and object.equals(CONSTANT) is that in is not needed to do null check in the first case, because if object is null we will just have false result of comparison :) The alternative to it can be something like 
if (object != null && object.equals(CONSTANT)). Can you please give your opinion what is the better way here?

**Comment by sanovikov71** [comment]
Hi, a new patch is ready.

**Review by lognaturel — APPROVED** [review]
